nil/internal/network/host.go

```diff
@@ -102,14 +114,18 @@ func newClient(ctx context.Context, conf *Config) (Host, error) {
 		var err error
 		privateKey, err = GeneratePrivateKey()
 		if err != nil {
-			return nil, err
+			return nil, zerolog.Nop(), err
 		}
 	}
 
-	options, err := getCommonOptions(ctx, privateKey)
+	options, logger, err := getCommonOptions(ctx, &Config{PrivateKey: privateKey})
 	if err != nil {
-		return nil, err
+		return nil, zerolog.Logger{}, err
```

Comment:
It's a bit inconsistent with return above: `zerolog.Logger{}` vs `zerolog.Nop()`